comp/rdnsquerier/impl/rdnsquerier_test.go

```diff
@@ -986,3 +993,81 @@ func TestCachePersist(t *testing.T) {
 	})
 	ts.validateExpected(t, expectedTelemetry)
 }
+
+func TestGetHostnameSync(t *testing.T) {
+	overrides := map[string]interface{}{
+		"network_devices.netflow.reverse_dns_enrichment_enabled": true,
+	}
+
+	ts := testSetup(t, overrides, true, nil, 0)
+	internalRDNSQuerier := ts.rdnsQuerier.(*rdnsQuerierImpl)
+
+	// Vic to reformat this test
+	// tests := []struct {
+	// 	description string
+	// 	ip          string
+	// 	expected    string
+	// 	errMsg      string
+	// }{
+	// 	{},
+	// }
+
+	ctx, cancel := context.WithTimeout(ts.ctx, 100*time.Second)
+	var hostname string
+	var err error
+
+	// // Test with invalid IP address
+	hostname, err = internalRDNSQuerier.GetHostnameSync(ctx, "invalid_ip")
+	assert.Error(t, err)
+	assert.Equal(t, "", hostname)
+
+	//Test with IP address not in private range
+	hostname, err = internalRDNSQuerier.GetHostnameSync(ctx, "8.8.8.8")
+	assert.Error(t, err)
+	assert.Equal(t, "", hostname)
+
+	// // Test with IP address in private range
+	hostname, err = internalRDNSQuerier.GetHostnameSync(ctx, "192.168.1.100")
+	assert.NoError(t, err)
+	assert.Equal(t, "fakehostname-192.168.1.100", hostname)
+
+	// Test with IP address in private range but cache miss, async look up succeeds
+	hostname, err = internalRDNSQuerier.GetHostnameSync(ctx, "192.168.1.101")
+	assert.NoError(t, err)
+	assert.Equal(t, "fakehostname-192.168.1.101", hostname)
```

Comment:
Is there a difference between this one and the above? Would we expect `192.168.1.100` to be a cache hit?